doc/changelog.md

```diff
@@ -244,6 +244,12 @@
       return y
   ```
 
+<h3>Documentation</h3>
+
+* A page has been added to the documentation, listing devices that are
+  Catalyst compatible.
+  [(#966)](https://github.com/PennyLaneAI/catalyst/pull/966)
```

Comment:
Do we need this docs changelog? So far I've always excluded docs, tests, and ci changes.

Reply:
Yeah I was a bit torn. I'm in favour of a docs section mostly because docs changes _do_ impact the source package someone downloads from PyPI and are impacted by (github) releases. Whereas CI/tests currently do not

Reply:
Note that we don't have source distributions on PyPI, although I suppose we do on Github. Tests are included in that source distribution as well though, only CI is excluded (I guess because the folder starts with a `.`, which just made me realize that those "source dists" are broken because some important files are excluded by this 😆)